flocker/node/_docker.py

```diff
@@ -527,6 +528,50 @@ def _address_in_use(self, apierror):
             return None
         return AddressInUse(address=(ip, int(port)), apierror=apierror)
 
+    def _image_data(self, image, container, state):
+        """
+        Supply data about an image, by either inspecting it or returning
+        cached data if available.
+
+        :param unicode image: The ID of the image.
+
+        :param unicode container: The ID of the container for which this
+            image is being used.
+
+        :param unicode state: The state of the container.
```

Comment:
Neither `container` nor `state` is necessary for the cache.  These parameters are only used for logging.  An alternative that makes this method simpler is to start a "looking up image data" action at the caller and log the container and state there.  Then the log message this method writes about the image not being found is written in a context that provides that information but without this method actually needing to take the extra arguments.